### PR TITLE
Normative: Adding Intl.Locale specification.

### PR DESCRIPTION
**Pull request by zbraniecki**
Adding https://github.com/tc39/proposal-intl-locale/ to ECMA402.

### CODE REVIEW

File: spec/locale.html [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+    <emu-clause id="sec-apply-options-to-tag" aoid=ApplyOptionsToTag>
+      <h1>ApplyOptionsToTag( _tag_, _options_ )</h1>
+      <p>
+        The following algorithm refers to <a href="https://www.unicode.org/reports/tr35/#Identifiers">UTS 35's Unicode Language and Locale Identifiers grammar</a>.
+      </p>
+
+      <emu-alg>
+        1. Assert: Type(_tag_) is String.
+        1. If IsStructurallyValidLanguageTag(_tag_) is *false*, throw a *RangeError* exception.
+        1. Let _language_ be ? GetOption(_options_, `"language"`, `"string"`, *undefined*, *undefined*).
```
Comment (caiolima) [inline spec/locale.html]: As @spectranaut also mentioned in my PR last week, now we are stylizing strings as values on ECMA402 and ECMA262 (see #396). This way, we need to use `*"language"*` or `*"string"*`

File: spec/locale.html [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+        The value of the [[RelevantExtensionKeys]] internal slot is &laquo; `"ca"`, `"co"`, `"hc"`, `"kf"`, `"kn"`, `"nu"` &raquo;. If %Collator%.[[RelevantExtensionKeys]] does not contain `"kf"`, then remove `"kf"` from %Locale%.[[RelevantExtensionKeys]]. If %Collator%.[[RelevantExtensionKeys]] does not contain `"kn"`, then remove `"kn"` from %Locale%.[[RelevantExtensionKeys]].
+      </p>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-properties-of-intl-locale-prototype-object">
+    <h1>Properties of the Intl.Locale Prototype Object</h1>
+
+    <p>
+      The Intl.Locale prototype object is itself an ordinary object. %LocalePrototype% is not an Intl.Locale instance and does not have an [[InitializedLocale]] internal slot or any of the other internal slots of Intl.Locale instance objects.
```
Comment (caiolima) [inline spec/locale.html]: We are missing `<dfn>%LocalePrototype%</dfn>`.

File: spec/locale.html (hunk 1 above)
Comment (spectranaut) [inline spec/locale.html]: In 402 and 262, we are stylizing strings are values now (in the past it was inconsistent): #396

So here you want:
```
         1. Let _language_ be ? GetOption(_options_, *"language"*, *"string"*, *undefined*, *undefined*).
```

I just tried locally, you can do a simple find and replace for ``` `" ``` -> ` *" ` and  ``` "` ``` -> ` "* ` in the `spec/locale.html` file and you will catch 'em all -- but see next comment first.

File: spec/locale.html [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+          1. Assert: _tag_ matches the `unicode_locale_id` production.
+          1. Set _tag_ to _tag_ with the substring corresponding to the `unicode_language_subtag` production replaced by the string _language_.
+        1. If _script_ is not *undefined*, then
+          1. If _tag_ does not contain a `unicode_script_subtag` production, then
+            1. Set _tag_ to the concatenation of the `unicode_language_subtag` production of _tag_, `"-"`, _script_, and the rest of _tag_.
+          1. Else,
+            1. Set _tag_ to _tag_ with the substring corresponding to the `unicode_script_subtag` production replaced by the string _script_.
+        1. If _region_ is not *undefined*, then
+          1. If _tag_ does not contain a `unicode_region_subtag` production, then
+            1. Set _tag_ to the concatenation of the `unicode_language_subtag` production of _tag_, the substring corresponding to the `"-" unicode_script_subtag` production if present, `"-"`, _region_, and the rest of _tag_.
```
Comment (spectranaut) [inline spec/locale.html]: The find+related option for string literal formatting will only break on this line. You should update this line first:

```
             1. Set _tag_ to the concatenation of the `unicode_language_subtag` production of _tag_, the substring corresponding to `"-"` and the `unicode_script_subtag` production if present, `"-"`, _region_, and the rest of _tag_.
```

File: spec/locales-currencies-tz.html [hunk 4]
```diff
@@ -54,13 +54,44 @@ <h1>IsStructurallyValidLanguageTag ( _locale_ )</h1>
       </p>
     </emu-clause>
 
-    <emu-clause id="sec-canonicalizelanguagetag" aoid="CanonicalizeLanguageTag">
```
Comment (spectranaut) [inline spec/locales-currencies-tz.html]: There are references to this now changed ESID in `annexes.html`, `locales-currencies-tz.html`, `negotiation.html` that need to be updated.

Comment (caiolima) [inline spec/locales-currencies-tz.html]: It seems that we still have some references to this old id.

File: spec/locales-currencies-tz.html [hunk 5]
```diff
@@ -54,13 +54,44 @@ <h1>IsStructurallyValidLanguageTag ( _locale_ )</h1>
...
+      <h1>CanonicalizeUnicodeLocaleId ( _locale_ )</h1>
 
       <p>
-        The CanonicalizeLanguageTag abstract operation returns the canonical and case-regularized form of the _locale_ argument (which must be a String value that is a structurally valid Unicode BCP 47 Locale Identifier as verified by the IsStructurallyValidLanguageTag abstract operation).
-        A conforming implementation shall take the steps specified in the &ldquo;BCP 47 Language Tag to Unicode BCP 47 Locale Identifier&rdquo; algorithm, from <a href="https://unicode.org/reports/tr35/#BCP_47_Language_Tag_Conversion">Unicode Technical Standard #35 LDML § 3.3.1 BCP 47 Language Tag Conversion</a>.
+        The CanonicalizeUnicodeLocaleId abstract operation returns the canonical and case-regularized form of the _locale_ argument (which must be a String value that is a structurally valid Unicode BCP 47 Locale Identifier as verified by the IsStructurallyValidLanguageTag abstract operation).
+        The following steps are taken:
       </p>
+
+<emu-alg>
```
Comment (caiolima) [inline spec/locales-currencies-tz.html]: nit: indentation seems not correct.

File: spec/locales-currencies-tz.html [hunk 6]
```diff
@@ -54,13 +54,44 @@ <h1>IsStructurallyValidLanguageTag ( _locale_ )</h1>
...
+        1. Let _localeId_ be the string _localeId_ after performing the algorithm to <a href="https://unicode.org/reports/tr35/#Canonical_Unicode_Locale_Identifiers">transform it to canonical form</a>.  (The result is a Unicode BCP 47 locale identifier, in both canonical syntax and canonical form.)
+        1. If _localeId_ contains a substring _extension_ that is a Unicode locale extension sequence, then
+          1. Let _components_ be ! UnicodeExtensionComponents(_extension_).
+          1. Let _attributes_ be _components_.[[Attributes]].
+          1. Let _keywords_ be _components_.[[Keywords]].
+          1. Let _newExtension_ be `"u"`.
```
Comment (caiolima) [inline spec/locales-currencies-tz.html]: Same comment about string literals being written as values now.

**Comment by zbraniecki** [comment]
Stage 4 has been granted as of February 5th 2020.

**Pull request merged**